inc/search.class.php

```diff
@@ -556,11 +549,12 @@ static function constructSQL(array &$data) {
       $searchopt        = &self::getOptions($data['itemtype']);
 
       $blacklist_tables = [];
+      $orig_table = ($data['itemtype'] == 'AllAssets' ? getTableForItemType($data['itemtype']) : $data['itemtype']::getTable());
```

Comment:
You may also break reservation search here, or anything that uses `$CFG_GLPI['union_search_type']`.
```suggestion
      $orig_table = (!is_a($data['itemtype'], CommonDBTM::class, true) ? getTableForItemType($data['itemtype']) : $data['itemtype']::getTable());
```

Reply:
OK, this need to be factorized :)